src/Microsoft.AspNet.Razor.Runtime/TagHelpers/HtmlTargetElementAttribute.cs

```diff
@@ -79,5 +79,11 @@ public HtmlTargetElementAttribute(string tag)
         /// </para>
         /// </remarks>
         public TagStructure TagStructure { get; set; }
+
+        /// <summary>
+        /// The required HTML element name of the direct parent.
+        /// </summary>
+        /// <remarks>A <c>null</c> value indicates any HTML element name is appropriate.</remarks>
```

Comment:
Oo, the `IsNullOrWhiteSpace` check is essentially just checking for whitespace in the characters. Prior to doing any character validation checks we validate if it's `null`